<?xml version="1.0"?>
<launch>

  <node pkg="relative_navigator" type="reference_trajectory_handler_node" name="reference_trajectory_handler_node" output="screen">
    <!-- <param name="model_path" value="$(find relative_navigator)/model/reference_trajectory_handler" /> -->
    <!-- <rosparam file="$(find relative_navigator)/config/param/reference_trajectory_handler.yaml" command="load"/> -->
  </node>
  <node pkg="relative_navigator" type="abstract_relative_pose_estimator_node.py" name="abstract_relative_pose_estimator_node" output="screen">
  </node>

  <node pkg="relative_navigator" type="local_goal_generator_node" name="local_goal_generator_node" output="screen">
  </node>

  <node pkg="ros_utils" type="odom_topic_to_tf_node" name="odom_topic_to_tf_node" output="screen">
  </node>

  <!-- <node pkg="ros_utils" type="odom_transformer_node" name="odom_transformer_node" output="screen"> -->
  <!-- </node> -->

  <node pkg="rviz" name="rviz_map_frame" type="rviz" args="-d $(find relative_navigator)/config/rviz/relative_navigator_map.rviz" output="screen"/>

  <node pkg="rviz" name="rviz_base_link_frame" type="rviz" args="-d $(find relative_navigator)/config/rviz/relative_navigator_base_link.rviz" output="screen"/>

  <node name="reference_image_republish" pkg="image_transport" type="republish" args="compressed raw">
      <remap from="in" to="/reference_image/image_raw/" />
      <remap from="out" to="/reference_image/image_raw/" />
  </node>
  <node name="observe_image_republish" pkg="image_transport" type="republish" args="compressed raw">
      <remap from="in" to="/usb_cam/image_raw/" />
      <remap from="out" to="/usb_cam/image_raw/" />
  </node>

</launch>

